src/openrct2/localisation/Formatting.cpp

```diff
@@ -792,6 +794,22 @@ namespace OpenRCT2
         BuildAnyArgListFromLegacyArgBuffer(fmt, anyArgs, args);
         return FormatStringAny(buffer, bufferLen, fmt, anyArgs);
     }
+
+    static void FormatMonthYear(std::stringstream& ss, int32_t month, int32_t year)
+    {
+        thread_local std::vector<FormatArg_t> tempArgs;
+        tempArgs.clear();
+
+        auto fmt = GetFmtStringById(STR_DATE_FORMAT_MY);
+        Formatter ft;
+        ft.Add<uint16_t>(month);
+        ft.Add<uint16_t>(year);
+        const void* legacyArgs = ft.Data();
+        BuildAnyArgListFromLegacyArgBuffer(fmt, tempArgs, legacyArgs);
```

Comment:
Do we need to use the legacy version?

Reply:
Do you see another way?

Reply:
Idk, open ended question, I'm assuming that if there's a legacy function, there is also a non legacy one

Reply:
There is and it causes said bug I am trying to fix. The reason is the non-legacy version doesn't support PUSH16 POP16, but it has no concept of argument byte size.